# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546478080 28800
#      Wed Jan 02 17:14:40 2019 -0800
# Branch stable
# Node ID bdcb5c1aaf4f2b4498f82f2a359e940afd51a5ba
# Parent  5455de7c8566b1a3b7aaf53653e553e26cc96d15
ov-base-scalar.h: Fix static analyzer detected issues (bug #55347).

* ov-base-scalar.h (issorted, is_sorted_rows): Check enum variable against an
enum value, rather than relying on default enum value being 0.

diff --git a/libinterp/octave-value/ov-base-scalar.h b/libinterp/octave-value/ov-base-scalar.h
--- a/libinterp/octave-value/ov-base-scalar.h
+++ b/libinterp/octave-value/ov-base-scalar.h
@@ -104,26 +104,26 @@ public:
                      sortmode) const
   {
     sidx.resize (dim_vector (1, 1));
     sidx(0) = 0;
     return octave_value (scalar);
   }
 
   sortmode issorted (sortmode mode = UNSORTED) const
-  { return mode ? mode : ASCENDING; }
+  { return mode == UNSORTED ? ASCENDING : mode; }
 
   Array<octave_idx_type> sort_rows_idx (sortmode) const
   {
     return Array<octave_idx_type> (dim_vector (1, 1),
                                    static_cast<octave_idx_type> (0));
   }
 
   sortmode is_sorted_rows (sortmode mode = UNSORTED) const
-  { return mode ? mode : ASCENDING; }
+  { return mode == UNSORTED ? ASCENDING : mode; }
 
   MatrixType matrix_type (void) const { return MatrixType::Diagonal; }
   MatrixType matrix_type (const MatrixType&) const
   { return matrix_type (); }
 
   bool is_scalar_type (void) const { return true; }
 
   bool isnumeric (void) const { return true; }
